### PR TITLE
Add reorg limit

### PR DESCRIPTION
**Pull request by SwenVanZanten**
As mentioned in #1059 the boys over at Firo had some ideas for this issue.
We should still check the numbers.

### CODE REVIEW

File: src/chainparams.cpp [hunk 1]
```diff
@@ -286,6 +286,10 @@ class CMainParams : public CChainParams {
             0.04004504697221486         
         };
 
+        // reorg
+        consensus.nMaxReorgDepth = 5;
+        consensus.nMaxReorgDepthEnforcementBlock = 388000;
```
Comment (justinvforvendetta) [inline src/chainparams.cpp:291]: hmm this block happened long ago, but that actually shouldn't affect anything.

**Comment by marpme** [comment]
**Suggestion:** For testing purposes you can lower the activation blockheight of this PR and check if it would successfully reject the longest chain that has attacked verge previously. It should correctly walk through the blocks and find the best chain even with the attacked chain is included. 

Ahh You also need to also remove the newer trusted blocks hashes.

File: src/chainparams.cpp [hunk 2]
```diff
@@ -283,12 +283,12 @@ class CMainParams : public CChainParams {
...
             1577552136,
             8015431,
-            0.04004504697221486         
+            0.04004504697221486
         };
 
         // reorg
         consensus.nMaxReorgDepth = 5;
-        consensus.nMaxReorgDepthEnforcementBlock = 388000;
+        consensus.nMaxReorgDepthEnforcementBlock = 4760000;
```
Comment (justinvforvendetta) [inline src/chainparams.cpp:291]: this is wayyy too high lol 
current block is 2549132, so maybe 2650000 is better or something like that.

Comment (justinvforvendetta) [reply src/chainparams.cpp:291]: nevermind. lol

**Comment by justinvforvendetta** [comment]
merged into v7 branch

**Pull request closed without merging**